pkg/k8s/pods/pod.go

```diff
@@ -76,19 +78,58 @@ func loopGetDevPod(dev *model.Dev, c *kubernetes.Clientset, waitUntilDeployed bo
 		return nil, err
 	}
 
-	log.Infof("pod %s with revison %s is progressing", pod.Name, d.Annotations[deploymentRevisionAnnotation])
-	if pod.Status.Phase == apiv1.PodRunning {
-		log.Infof("pod %s with revison %s is running", pod.Name, d.Annotations[deploymentRevisionAnnotation])
-		return pod, nil
+	opts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", OktetoInteractiveDevLabel, dev.Name),
+		FieldSelector: fmt.Sprintf("metadata.name=%s", pod.Name),
+	}
+
+	watchPod, err := c.CoreV1().Pods(dev.Namespace).Watch(opts)
+	if err != nil {
+		return nil, err
 	}
-	for _, status := range pod.Status.ContainerStatuses {
-		if status.State.Waiting != nil {
-			if status.State.Waiting.Reason == "ErrImagePull" || status.State.Waiting.Reason == "ImagePullBackOff" {
-				return nil, fmt.Errorf(status.State.Waiting.Message)
+	opts = metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("involvedObject.kind=Pod,involvedObject.name=%s", pod.Name),
+	}
+	watchPodEvents, err := c.CoreV1().Events(dev.Namespace).Watch(opts)
+	if err != nil {
+		return nil, err
+	}
+	for {
+		select {
+		case event := <-watchPod.ResultChan():
+			pod, ok := event.Object.(*v1.Pod)
+			if !ok {
+				return nil, nil
 			}
+			if pod.Status.Phase == apiv1.PodRunning {
+				return pod, nil
+			}
+			if pod.DeletionTimestamp != nil {
```

Comment:
This would mean that the pod is in process of being deleted. What happens in this case? The caller of GetDevPod will most likely get a nullref no? (since there's no error returned)